app/misc/git.ts

```diff
@@ -495,6 +495,56 @@ function createBranch() {
   }
 }
 
+// search for tags
+function searchTag() {
+  Git.Repository.open(repoFullPath)
+    .then(function (repo) {
+      repo.getCurrentBranch()
+        .then(function () {
+          // grab the list of references - these could be branches or tags
+          return repo.getReferences(Git.Reference.TYPE.LISTALL);
+        }).then(function (refList) {
+          refList.sort();
+          bname = {};
```

Comment:
Why are these objects getting cleared?

Reply:
And couldnt you just get the tag name from the tags object? that way we dont need to continue querying node git

Reply:
I wasn't sure if the branches and tags would be reloaded in real time. This only queries nodegit if the tags are being searched for. It's just to ensure we have the latest copy of tags/branches. But you're right that it would be more efficient to use the tags object, if it's up to date. I can test it out with the tags object and if it's working well, I submit a changeset.

Reply:
It probably doesn't matter too much, the object should be up to date though. Regardless, they shouldn't be cleared from within this function, thats probably more important.